lib/purserl.ex

```diff
@@ -220,20 +220,44 @@ defmodule Purserl do
 
     verbose = not Enum.member?(["", "0", "false"], System.get_env("PURERLEX_VERBOSE", ""))
 
+    max_length =
+        case :io.columns() do
+          {:ok, cols} -> cols
+          {:error, :enotsup} -> 10000
+        end
+
+    # NOTE[em]: It is a bit complex to calculate what should be output since
+    # the IO device may shrink and we also need to color some parts. Colors
+    # don't take up IO device width.
+    output_row =
+      [step_in_brackets, " ", s_version, " ", label, " ", module]
+      |> Enum.reduce({max_length, []}, fn s, {remaining, acc} ->
+        sliced = String.slice(s, 0, remaining)
+        out =
+          case s do
+            ^label -> Color.format([color, sliced])
+            _ -> sliced
+          end
+        {remaining - String.length(sliced), [out | acc]}
+      end)
+      |> elem(1)
+      |> Enum.reverse()
+
```

Comment:
why?

Reply:
The output sometimes breaks when using a small window. This is to prevent that

Reply:
in what way does it break? how does this help?

Reply:
We rely on the fact that we only print one row when overwriting rows, but sometimes when the window is small enough the output will become two rows instead, which means more rows of input when only one is desired (not verbose) and rows becoming jumbled when updating several rows during compilation (verbose)

Reply:
cool, okey, thanks. There aren't any ansi codes for moving around logical lines rather than visual lines?

Reply:
(I think I've seen this break already when I scroll around while it's still updating the verbose output)

Reply:
> cool, okey, thanks. There aren't any ansi codes for moving around logical lines rather than visual lines?

Not that I know about. I think it is restricted to manipulating a cursor. I'm not sure how scrolling affects the output, but it will likely break then as well. This should at least fix the issue if the window is not manipulated during a run

Reply:
anything that scrolls off screen is impossible to track using this method then, which is pretty bad. I have quite a small font size, but it's not small enough for 1k line terminals

Reply:
now that we've gotten through the why, perhaps you were looking for https://hexdocs.pm/elixir/1.12/Enum.html#chunk_every/2 ? either way, ship it

Reply:
I actually think this is one of the few cases which require a fold, but perhaps there is something I overlooked